### ISSUE TITLE
document & *test* asymptotic complexity of collection operations

### ISSUE THREAD

**Issue by adriaanm** [scala/scala-dev#452]
I believe there was a framework by @axel22 that does this. Was it scalameter?

See also http://docs.scala-lang.org/overviews/collections/performance-characteristics.html, which should be updated, rendered in the source and then enforced using automated regression testing.

**Comment by axel22** [comment]
No, ScalaMeter does not do this by default, its main goal is measuring absolute performance numbers.

**Comment by adriaanm** [comment]
You're right. I was trying to remember which tool measures/tests asymptotic complexity and your name popped up. Have you perhaps authored something else that does this? :-) Or does it ring a bell?

**Comment by axel22** [comment]
No, I didn't do anything like that. I think @jsuereth did something in that direction. Or maybe Rex Kerr? Not sure. :/

**Comment by adriaanm** [comment]
Ok, thanks! Rex did https://github.com/Ichoran/thyme, but I don't think that does O(n) either. Maybe I'm just wishful remembering :-)
